### PR TITLE
Fix CPMS API version

### PR DESCRIPTION
**Pull request by clobrano**
MDR currently looks for Machine Owner under
machine.openshift.api/v1beta1 API, while ControlPlaneMachineSet are
defined under machine.openshift.api/v1 API.

- include machine.openshift.api/v1 package, and update related packages
- extend getMachineOwner to select the appropriate API for the given
  Machine Owner Kind

fixes: https://issues.redhat.com/browse/ECOPROJECT-1595

Signed-off-by: Carlo Lobrano <c.lobrano@gmail.com>

### CODE REVIEW

File: controllers/suite_test.go [hunk 1]
```diff
@@ -144,7 +143,8 @@ var _ = BeforeSuite(func() {
 		Expect(err).ToNot(HaveOccurred())
 	}()
 
-	v1beta1.AddToScheme(cclient.Scheme())
+	machinev1.AddToScheme(cclient.Scheme())
```
Comment (razo7) [inline controllers/suite_test.go:146]: NIT: Maybe surround these two `AddToScheme` calls with Expect(DO_SOMETHING).ToNot(HaveOccurred())

**Review by razo7 — APPROVED** [review]
Left a question and a nit

File: main.go [hunk 2]
```diff
@@ -75,10 +78,13 @@ func main() {
...
 
+	metricOpts := server.Options{
+		BindAddress: metricsAddr,
+	}
+
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
+		Metrics:                metricOpts,
```
[the remark this answers is not in the document]
Comment (clobrano) [reply main.go:87]: This is due to `sigs.k8s.io/controller-runtime` update to v0.16.1, which in turn was needed to fix this error

```
go vet ./...
# sigs.k8s.io/controller-runtime/pkg/cache
vendor/sigs.k8s.io/controller-runtime/pkg/cache/multi_namespace_cache.go:308:9: cannot use handles (variable of type map[string]"k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration) as "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration value in return statement: map[string]"k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration does not implement "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration (missing method HasSynced)
vendor/sigs.k8s.io/controller-runtime/pkg/cache/multi_namespace_cache.go:321:9: cannot use handles (variable of type map[string]"k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration) as "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration value in return statement: map[string]"k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration does not implement "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration (missing method HasSynced)
vendor/sigs.k8s.io/controller-runtime/pkg/cache/multi_namespace_cache.go:326:17: impossible type assertion: h.(map[string]toolscache.ResourceEventHandlerRegistration)
        map[string]"k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration does not implement "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration (missing method HasSynced)
make: *** [Makefile:136: vet] Error 1
```

**Pull request merged**